src/Lykke.Service.ExchangeDataStore/Lykke.Service.ExchangeDataStore.AzureRepositories/OrderBooks/OrderBookSnapshotEntity.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Lykke.Service.ExchangeDataStore.Core.Helpers;
+using System;
+
+namespace Lykke.Service.ExchangeDataStore.AzureRepositories.OrderBooks
+{
+    public sealed class OrderBookSnapshotEntity: BaseEntity
+    {
+        public string UniqueId => $"{PartitionKey}_{RowKey}";
+
+        public string Exchange { get; }
+
+        public string AssetPair { get; }
+
+        public DateTime SnapshotDateTime { get; }
+
+        public OrderBookSnapshotEntity()
+        {
+        }
+
+        public OrderBookSnapshotEntity(string exchange, string assetPair, DateTime snapShotTimestamp)
+        {
+            SnapshotDateTime = snapShotTimestamp;
+            Exchange = exchange;
+            AssetPair = assetPair;
+
+            PartitionKey = $"{exchange}_{assetPair}".RemoveSpecialCharacters('-', '_', '.');
+            RowKey = snapShotTimestamp.ToSnapshotTimestampFormat();
```

Comment:
Ok, so I assume you are suggesting to use a global static counter as a suffix to RowKey, instead of extending the timestamp precision to five digits for the millisecond part (fffff) which is how i did it to ensure uniqueness. That's fine with me, if this is what you meant here - I have added the static reverted timestamp format to 3 fs (fff).

Reply:
The problem is the clock on the server can be corrected and moved back and it is a very small chance (but not 0) that you get the same timestamp twice. With the counter even if the timestamp the same key will not have doubles.

Reply:
Got it, ok, thanks. :)

Reply:
Done.